packages/@aws-cdk/aws-ecs/lib/container-image.ts

```diff
@@ -33,12 +44,26 @@ export abstract class ContainerImage {
    */
   public abstract readonly imageName: string;
 
+  /**
+   * Optional credentials for a private image registry
+   */
+  public abstract readonly credentials?: RepositoryCreds;
```

Comment:
Does this need to be publicly accessible? Seems like `renderRepositoryCredentials()` is all that a consumer of this class should need from it, right?

Reply:
[`renderContainerDefinition()`](https://github.com/allisaurus/aws-cdk/pull/1/files#diff-541fb20ec37d9e651dd38abc923ccb58R382) needs to see it to determine whether or not to call `renderRepositoryCredentials()`. ECS will fail to register the task definition if `RepositoryCredentials` is present but empty.